server/src/com/cloud/network/router/VirtualNetworkApplianceManagerImpl.java

```diff
@@ -2587,7 +2589,13 @@ public boolean postStateTransitionEvent(final StateMachine2.Transition<VirtualMa
                 event == VirtualMachine.Event.FollowAgentPowerOnReport &&
                 newState == VirtualMachine.State.Running &&
                 isOutOfBandMigrated(opaque)) {
-            s_logger.debug("Virtual router " + vo.getInstanceName() + " is powered-on out-of-band");
+            /* Since vRouter appears to be powered-on OOB, make sure we can talk to router
+             * If we can't talk to it, we need to reboot it to get it managed correctly
+             * This is needed for example when a host agent goes down and comes back up,
+             * we would have done a failed HA event on the router and end up having our controlIP out-of-sync
+             */
+             s_logger.info("Router " + vo.getInstanceName() + " (ID:" + vo.getId() + ") is powered-on out-of-band, checking if can send CheckRouterCommand to router");
```

Comment:
:+1: completely agree with this change, the ecosystem shouldn't rely on debug logging anyway. friendliness to the community dictates diligent testing anyway.